src/settings-ui/Settings.UI/ViewModels/ShellViewModel.cs

```diff
@@ -76,6 +77,10 @@ public void Initialize(Frame frame, NavigationView navigationView, IList<Keyboar
             NavigationService.NavigationFailed += Frame_NavigationFailed;
             NavigationService.Navigated += Frame_Navigated;
             this.navigationView.BackRequested += OnBackRequested;
+
+            var topLevelItems = navigationView.MenuItems.OfType<NavigationViewItem>().ToArray();
+
+            _fullListOfNavViewItems = topLevelItems.Union(topLevelItems.SelectMany(menuItem => menuItem.MenuItems.OfType<NavigationViewItem>())).ToArray();
```

Comment:
We need to ensure it's only allocating the `_fullListOfNavViewItems` if it's null, otherwise it will run through every single time the page is initialized.

```suggestion
            if (_fullListOfNavViewItems == null)
            {
                 var topLevelItems = navigationView.MenuItems.OfType<NavigationViewItem>();
                 _fullListOfNavViewItems = topLevelItems.Union(topLevelItems.SelectMany(menuItem => menuItem.MenuItems.OfType<NavigationViewItem>())).ToArray();
            }
```